oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/http/HTTPServer.java

```diff
@@ -70,6 +78,21 @@ public void initialize() {
             })
             .decorator(LoggingService.newDecorator());
 
+        if (config.isEnableTLS()) {
+            sb.https(config.getHttpsPort());
+            if (config.isEnableTlsSelfSigned()) {
+                sb.tlsSelfSigned();
```

Comment:
It's impossible to use self signed certificate for services that are exposed to external consumer, i.e. aws won't definitely skip verify certificate like you do in the unit tests, so I think this is useless in reality.